src/components/UserForm.js

```diff
@@ -86,17 +73,17 @@ const AddUserForm = (props) => {
 		}
 
 		//preventing data submit
-		if (error.isGenderValid === false || error.isFnameValid === false) {
+		if (error.isGenderValid === false || formData.fname.length < 3) {
 			return;
 		}
 
 		const userData = {
-			fname: fname,
-			lname: lname,
-			email: email,
-			country: country,
-			dob: dob,
-			gender: gender,
+			fname: formData.fname,
+			lname: formData.lname,
+			email: formData.email,
+			country: formData.country,
+			dob: formData.dob,
+			gender: formData.gender,
 		};
 
 		props.regData(userData);
```

Comment:
Why you are creating another variable useData, you can directly pass formData if there are no changes in keys